Sources/_NIOConcurrency/AsyncAwaitSupport.swift

```diff
@@ -62,9 +62,11 @@ extension EventLoopPromise {
     ///
     /// - parameters:
     ///   - body: The `async` function to run.
+    /// - returns: A `Task` which was created to `await` the `body`.
     @available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+    @discardableResult
     @inlinable
-    public func completeWithAsync(_ body: @escaping () async throws -> Value) {
+    public func completeWithTask(_ body: @escaping () async throws -> Value) -> Task<Void, Never> {
```

Comment:
Should we deprecate and replace given we're changing the name of the function?

Reply:
We don't have a stable API in the _NIOConcurrency module. Hence the underscore.

Reply:
Sure, but it's a less painful break and we can always remove them later.

Reply:
Eh, I don't think it matters much, we've messed about with these a whole lot.

Reply:
Fair enough